src/SudoSu.php

```diff
@@ -23,15 +23,27 @@ public function __construct(Application $app, AuthManager $auth, SessionManager
         $this->session = $session;
     }
 
-    public function loginAsUser($userId, $currentUserId)
+    /**
+     * Log the sudosu user in and preserve the original user id if exists
+     *
+     * @param string $sudosuUserId
+     * @param string|null $originalUserId
+     * @return void
+     */
+    public function loginAsUser($sudosuUserId, $originalUserId)
     {
         $this->session->put('sudosu.has_sudoed', true);
-        $this->session->put($this->sessionKey, $currentUserId);
+        $this->session->put($this->sessionKey, $originalUserId);
 
-        $this->auth->loginUsingId($userId);
+        $this->auth->loginUsingId($sudosuUserId);
```

Comment:
@mrterryh see this link https://www.python.org/dev/peps/pep-0020/#id3 (Explicit is better than implicit)

also read this article  (https://blog.goyello.com/2013/05/17/express-names-in-code-bad-vs-clean/) 

as $userId and $currentUserId wasn't clear enough what's the role of each id and made me confused when i was reading the code .

one last thing , i recommend this book especially first six chapters 
https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882